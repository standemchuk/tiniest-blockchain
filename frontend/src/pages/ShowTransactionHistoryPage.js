import React, { Component } from 'react'
import Grid from '@material-ui/core/Grid'

import ErrorModal from '../components/ErrorModal'
import GenericTable from '../components/GenericTable'
import PageHeader from '../components/PageHeader'

export default class ShowTransactionHistoryPage extends Component {
  state = {
    transactions: [],
    requestError: false
  }

  async componentDidMount () {
    await this.getTransactions()
  }

  getTransactions = async () => {
    try {
      const response = await fetch('http://localhost:3000/api/transactions')

      if (response && response.status === 200) {
        const responseJson = await response.json()

        this.setState({ transactions: responseJson.transactions })
      } else {
        this.setState({ requestError: true })
      }
    } catch (err) {
      this.setState({ requestError: true })
    }
  }

  handleRetry = async () => {
    await this.getTransactions()
  }

  handleModalClose = () => {
    this.setState({ requestError: false })
  }

  render () {
    return (
      <Grid container>
        <PageHeader title='Transaction history' />
        <Grid item xs={12}>
          <GenericTable
            data={this.state.transactions}
            headers={['Sender', 'Receiver', 'Amount', 'Created at']}
          />
          <ErrorModal
            open={this.state.requestError}
            handleModalClose={this.handleModalClose}
            handleRetry={this.handleRetry}
          />
        </Grid>
      </Grid>
    )
  }
}
