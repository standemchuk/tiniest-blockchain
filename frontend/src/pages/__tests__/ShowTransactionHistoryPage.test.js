import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import ShowTransactionHistoryPage from '../ShowTransactionHistoryPage'

beforeEach(() => {
  fetch.resetMocks()
})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ShowTransactionHistoryPage />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should fetch the data and set requestError to false if successful', () => {
  const tree = shallow(
    <ShowTransactionHistoryPage />
  )

  expect(fetch).toBeCalled()
  expect(tree.state().requestError).toBe(false)
})
