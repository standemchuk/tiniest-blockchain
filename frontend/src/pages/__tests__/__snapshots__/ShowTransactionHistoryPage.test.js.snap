// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-container-1"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-40"
  >
    <p
      className="MuiTypography-root-98 MuiTypography-display1-102 MuiTypography-paragraph-116 MuiTypography-alignCenter-111 Component-header-97"
    >
      Transaction history
    </p>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-40"
  >
    <div
      className="MuiGrid-container-1"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-40"
      >
        <h3
          className="MuiTypography-root-98 MuiTypography-subheading-105"
          style={
            Object {
              "marginLeft": 30,
            }
          }
        >
          No data
        </h3>
      </div>
    </div>
  </div>
</div>
`;
